src/main/java/frc/robot/RobotContainer.java

```diff
@@ -73,6 +76,10 @@ public RobotContainer() {
       controlBindings = new JoystickControlBindings();
     }
 
+    NamedCommands.registerCommand("aimAndShoot", autoBuilder.aimAndShoot());
+    NamedCommands.registerCommand("shootDonut", autoBuilder.shootDonut());
+    NamedCommands.registerCommand("startIntake", autoBuilder.startIntake());
```

Comment:
I haven't thought too much about this, but it seems like this probably belongs in AutonomousBuilder, I think named commands are its purpose.

Reply:
OK, yeah I see that, it says it's recommended to register them in RobotContainer. PP's example code is mostly using [instance factory methods](https://docs.wpilib.org/en/stable/docs/software/commandbased/organizing-command-based.html#instance-command-factory-methods). I messed around with this in my robot 1 code, but I found it to be restrictive since we have so many commands that span subsystems.

I'm open to discussing this, but since you already created a [non-static command factory](https://docs.wpilib.org/en/stable/docs/software/commandbased/organizing-command-based.html#non-static-command-factories), and that has worked well for us in the past, I thought we would probably continue with that design. If we do, I think it makes sense to have the factory responsible for registering the commands. This way, the addition or removal of a command only requires a change in one class.

Something that programmers talk a lot about is ["The Single Responsibility Principle"](https://en.wikipedia.org/wiki/Single_responsibility_principle) which means "Gather together the things that change for the same reasons. Separate those things that change for different reasons."